stdlib/public/Concurrency/TaskSleep.swift

```diff
@@ -107,23 +103,18 @@ extension Task where Success == Never, Failure == Never {
 
   /// Called when the sleep(nanoseconds:) operation woke up without being
   /// canceled.
-  static func onSleepWake(
-      _ wordPtr: UnsafeMutablePointer<Builtin.Word>
-  ) {
+  static func onSleepWake(_ token: borrowing Atomic<SleepState>) {
     while true {
-      let state = SleepState(loading: wordPtr)
+      let state = token.load(ordering: .relaxed)
```

Comment:
I haven't yet thought about whether the load ordering values used in this PR are correct.

Reply:
AFAICS there isn't any other state that depends on this synchronization being sequential (as the original code was); we just swap the state and take its continuation; the other methods called should not need any synchronization (task creation or continuation resumption). So it seems to me relaxed should be ok here